### PR TITLE
Fixed totally incorrect docblocks

### PR DESCRIPTION
**Pull request by GrahamCampbell**
For example, `Integer` implies only an object of type `Integer` is allowed, but what you really meant was only a variable of a primitive type `int` is allowed.

### CODE REVIEW

File: phpseclib/Crypt/Base.php [hunk 1]
```diff
@@ -1398,14 +1398,14 @@ function _clearBuffers()
     }
 
     /**
-     * String Shift
```
Comment (bantu) [inline phpseclib/Crypt/Base.php]: ?

File: phpseclib/Crypt/Hash.php [hunk 2]
```diff
@@ -823,13 +823,13 @@ function _add()
     }
 
     /**
-     * String Shift
+     * string Shift
```
Comment (bantu) [inline phpseclib/Crypt/Hash.php]: ?

File: phpseclib/Crypt/RSA.php [hunk 3]
```diff
@@ -142,11 +142,11 @@
  */
 define('CRYPT_RSA_ASN1_INTEGER',     2);
 /**
- * ASN1 Bit String
+ * ASN1 Bit string
```
Comment (bantu) [inline phpseclib/Crypt/RSA.php]: ?

File: phpseclib/Crypt/RSA.php [hunk 4]
```diff
@@ -142,11 +142,11 @@
...
  */
 define('CRYPT_RSA_ASN1_BITSTRING',   3);
 /**
- * ASN1 Octet String
+ * ASN1 Octet string
```
Comment (bantu) [inline phpseclib/Crypt/RSA.php]: ?

File: phpseclib/Crypt/RSA.php [hunk 5]
```diff
@@ -1907,13 +1907,13 @@ function _encodeLength($length)
     }
 
     /**
-     * String Shift
+     * string Shift
```
Comment (bantu) [inline phpseclib/Crypt/RSA.php]: ?

File: phpseclib/File/ASN1.php [hunk 6]
```diff
@@ -1234,13 +1234,13 @@ function loadFilters($filters)
     }
 
     /**
-     * String Shift
+     * string Shift
```
Comment (bantu) [inline phpseclib/File/ASN1.php]: ?

File: phpseclib/File/ASN1.php [hunk 7]
```diff
@@ -1251,15 +1251,15 @@ function _string_shift(&$string, $index = 1)
     }
 
     /**
-     * String type conversion
+     * string type conversion
```
Comment (bantu) [inline phpseclib/File/ASN1.php]: ?

**Comment by bantu** [comment]
This PR looks pretty good to me. I've marked a few occurrences of "string" which should stay as "String". Please ensure these and following are changed back.

**Comment by bantu** [comment]
I agree it should be "bool" instead of "boolean" and "int" instead of "integer". We have to write that down somewhere.

File: phpseclib/File/ASN1.php (hunk 6 above)
Comment (GrahamCampbell) [inline phpseclib/File/ASN1.php]: oops...

**Comment by GrahamCampbell** [comment]
@bantu Would you mind finishing this off please. Sorry I haven't had time to get back around to this.

**Comment by GrahamCampbell** [comment]
I'm going to start again with this...

**Pull request closed without merging**